packages/truffle-compile/compilerSupplier/loadingStrategies/LoadingStrategy.js

```diff
@@ -15,11 +15,17 @@ class LoadingStrategy {
     };
     this.config = Object.assign({}, defaultConfig, options);
 
-    const compilerCachePath = path.resolve(
+    const compilersDir = path.resolve(
       Config.getTruffleDataDirectory(),
       "compilers"
     );
-    if (!fs.existsSync(compilerCachePath)) fs.mkdirSync(compilerCachePath);
+    const compilerCachePath = path.resolve(compilersDir, "node_modules"); // because babel binds to require & does weird things
+    if (!fs.existsSync(compilersDir)) {
+      fs.mkdirSync(compilersDir);
+      fs.mkdirSync(compilerCachePath);
```

Comment:
This, however, I think you can remove. The logic is also more direct there, just two one-liners.

Reply:
I'm not sure I follow?

If the `compilersDir` doesn't already exist, then create it & then create the `node_modules` folder.

If it does already exist, but the full path doesn't already exist (5.0.8 users using BYOC will already have a `compilers` dir), then make sure it does by creating the `node_modules` folder.

Reply:
If I'm missing something, feel free to use the github code suggestion thingy.

Reply:
I'm saying 
```suggestion
      if (!fs.existsSync(compilersDir)) fs.mkdirSync(compilerDir);
      if (!fs.existsSync(compilerCachePath)) fs.mkdirSync(compilerCachePath);
```
I mean yours will work but this is just simpler.

Reply:
What I mean to say is that it would be simpler if the `node_modules` directory were only created in the step that checks to see if it exists. So the logic is:
  1. If the compilers directory doesn't exist, create it.
  2. If the cache directory doesn't exist, create it.

Reply:
Oh! I see. Cool 😎